sklearn/linear_model/omp.py

```diff
@@ -806,6 +806,7 @@ def fit(self, X, y):
             returns an instance of self.
         """
         X, y = check_X_y(X, y)
+        X = as_float_array(X, copy=False)
```

Comment:
That behavior would probably be good, but not sure if it would be identical with ``as_float_array`` which also checks the order.